### PR TITLE
style(ui): typography overhaul

### PR DESCRIPTION
**Pull request by juliendonck**
This pr aligns the design system in upstream with what we have in [Figma](https://www.figma.com/file/5mBBm9yuRRuIUk3pCgC3HZ/%F0%9F%94%A1-Typography?node-id=2%3A0).

### How this should be used (also in [DEVELOPMENT.md](https://github.com/radicle-dev/radicle-upstream/pull/749/files#diff-f785e9e46b9afd2567df7a658338db17R268))

The design system provides a constrained set of typographic styles. This consists of a set of styled headers, a set of styled paragraphs and a set of modifiers. These also overlap with the components we have in our design system in Figma, where the design of the app exists. All classes are prefixed with `typo-` so this might be helpful if you have any autocomplete in your editor.

For the headers you can just use `<h1>` up to `<h5>`, if you want to apply the same styles to other html elements you can use the matching classes `typo-header-1` to `typo-header-5` (use `<h1>` to `<h5>` where you can).

For text we you can use the classes that start with `typo-text`. These come in 2 sizes, the normal one and `typo-text-small`. Check out [typography.css](./public/typography.css) to get an idea of the possible combinations. All the ones we're using in Figma are represented here.

The modifiers give us some flexibility and allow us to create classes for certain css functionality we use over and over. Such as, `typo-overflow-ellipses` and `typo-all-caps`. These should be self-explanatory.

We also added a set of modifiers that allow you to add the font-family as a class where you need it, here again we would recommend not doing that as most styles should fit into one of the two categories above.

The only place in the app where we're not using this is in `<Markdown />`, since the library we use doesn't allow us to overwrite the styles without using global declarations. If you have any questions or improvements, open an issue and we're
happy to help you along.

### CODE REVIEW

File: ui/Screen/DesignSystemGuide.svelte [hunk 1]
```diff
@@ -356,79 +341,88 @@
...
-          border-radius: 2px; padding: 4px 8px;">
-          {color}
-        </Text>
-      {/each}
-    </Section>
+    <h1 style="margin-bottom: 92px">Primitives</h1>
 
     <Section
-      title="Typography"
+      title="Typography updated"
```
Comment (rudolfs) [inline ui/Screen/DesignSystemGuide.svelte:347]:
```suggestion
      title="Typography"
```

Once the PR is merged this will be the de facto typography.

**Comment by MeBrei** [comment]
The user name in the transaction detail view/summary is now in all caps.
Screenshot 2020-08-05 at 15 22 43

File: ui/DesignSystem/Component/SourceBrowser/RevisionSelector.svelte [hunk 2]
```diff
@@ -48,6 +48,14 @@
 </script>
 
 <style>
+  .revision-name {
+    color: var(--color-foreground-level-6);
+    white-space: nowrap;
+    text-overflow: ellipsis;
```
[the remark this answers is not in the document]
Comment (juliendonck) [reply ui/DesignSystem/Component/SourceBrowser/RevisionSelector.svelte]: https://github.com/radicle-dev/radicle-upstream/pull/749/commits/8c2c001f3b4c97228fa49fa4bbb961110c4d264c

File: ui/DesignSystem/Component/ProjectCard.svelte [hunk 3]
```diff
@@ -30,23 +30,28 @@
     display: flex;
     margin-left: 9px;
   }
+
+  .desc {
+    color: var(--color-foreground-level-5);
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
+  }
```
[the remark this answers is not in the document]
Comment (juliendonck) [reply ui/DesignSystem/Component/ProjectCard.svelte:36]: https://github.com/radicle-dev/radicle-upstream/pull/749/commits/8c2c001f3b4c97228fa49fa4bbb961110c4d264c

File: ui/DesignSystem/Component/Dropdown/Option.svelte [hunk 4]
```diff
@@ -50,9 +50,9 @@
       {...avatarProps}
       {disabled} />
   {:else}
-    <Text
+    <p
       style={`overflow:hidden; text-overflow: ellipsis; margin: 0 42px  0 12px; color: ${disabledColor}`}>
```
[the remark this answers is not in the document]
Comment (juliendonck) [reply ui/DesignSystem/Component/Dropdown/Option.svelte]: https://github.com/radicle-dev/radicle-upstream/pull/749/commits/8c2c001f3b4c97228fa49fa4bbb961110c4d264c

File: ui/DesignSystem/Component/SourceBrowser/CommitTeaser.svelte [hunk 5]
```diff
@@ -48,28 +46,31 @@
...
+    <p class="commit-message typo-small">{commitMessage}</p>
   </div>
 
   <div class="align-right">
-    <UserCard
-      {user}
-      style="margin-right: 8px; color: var(--color-foreground-level-6)" />
-    <Text style="color: var(--color-foreground-level-6)">{timestamp}</Text>
+    <p
+      class="small-bold"
```
[the remark this answers is not in the document]
Comment (juliendonck) [reply ui/DesignSystem/Component/SourceBrowser/CommitTeaser.svelte]: https://github.com/radicle-dev/radicle-upstream/pull/749/commits/abdbf69094d124645edfb0f360e65c301b6ab714

File: ui/DesignSystem/Component/Transaction.svelte [hunk 6]
```diff
@@ -32,13 +32,17 @@
   $: updateAvatar();
 </script>
 
+<style>
+  p {
```
[the remark this answers is not in the document]
Comment (juliendonck) [reply ui/DesignSystem/Component/Transaction.svelte]: https://github.com/radicle-dev/radicle-upstream/pull/749/commits/08fce83d342a86eaf70a5686933ddc0b47de1468

File: DEVELOPMENT.md [hunk 7]
```diff
@@ -266,6 +265,40 @@ use slots:
...
+
+For the headers you can just use `<h1>` up to `<h5>`, if you want to apply the
+same styles to other html elements you can use the matching classes
+`typo-header-1` to `typo-header-5` (use `<h1>` to `<h5>` where you can).
+
+For text we recommend using a `<p>` wherever they are used in the Figma files.
+There are matching modifiers if you're looking for a bold paragraph, `<p
+class="typo-bold">`, or any of the other ones we have. Check out
+[typography.css](./public/typography.css) to get an idea of the possible
+combinations. All the ones we're using in figma are represented here. If you
```
[the remark this answers is not in the document]
Comment (juliendonck) [reply DEVELOPMENT.md]: https://github.com/radicle-dev/radicle-upstream/pull/749/commits/c29579df6a3327286cf0c3bf44400fbf8f417972

File: DEVELOPMENT.md [hunk 8]
```diff
@@ -266,6 +265,40 @@ use slots:
...
+need to use another html element than `<p>`, there are classes that match the
+paragraph styling, e.g.: `<p class="typo-bold">` -> `<span
+class="typo-text-bold">`
+
+The modifiers are to give us some flexibility and allow us to create classes for
```
[the remark this answers is not in the document]
Comment (juliendonck) [reply DEVELOPMENT.md]: https://github.com/radicle-dev/radicle-upstream/pull/749/commits/d835bd325c45ec5fcd6727862265a7ba9ebfdd42

File: DEVELOPMENT.md [hunk 9]
```diff
@@ -266,6 +265,40 @@ use slots:
...
+certain css functionality we use over and over. Such as,
+`typo-overflow-ellipses` and `typo-all-caps`. These should be self-explanatory.
+
+We also added a set of modifiers that allow you to add the font-family as a
+class where you need it, here again we would recommend not doing that as most
+styles should fit into one of the two categories above.
+
+The only place in the app we're not using this is in `<Markdown />`, since the
```
[the remark this answers is not in the document]
Comment (juliendonck) [reply DEVELOPMENT.md]: https://github.com/radicle-dev/radicle-upstream/pull/749/commits/d835bd325c45ec5fcd6727862265a7ba9ebfdd42

File: DEVELOPMENT.md [hunk 10]
```diff
@@ -266,6 +265,40 @@ use slots:
...
+`typo-header-1` to `typo-header-5` (use `<h1>` to `<h5>` where you can).
+
+For text we recommend using a `<p>` wherever they are used in the Figma files.
+There are matching modifiers if you're looking for a bold paragraph, `<p
+class="typo-bold">`, or any of the other ones we have. Check out
+[typography.css](./public/typography.css) to get an idea of the possible
+combinations. All the ones we're using in figma are represented here. If you
+need to use another html element than `<p>`, there are classes that match the
+paragraph styling, e.g.: `<p class="typo-bold">` -> `<span
+class="typo-text-bold">`
```
[the remark this answers is not in the document]
Comment (juliendonck) [reply DEVELOPMENT.md]: https://github.com/radicle-dev/radicle-upstream/pull/749/commits/26f88a50aebb03ddb3b2973047f57a4bd486d276

File: ui/Screen/DesignSystemGuide.svelte (hunk 1 above)
Comment (juliendonck) [reply ui/Screen/DesignSystemGuide.svelte]: improve design system layout for typography

Comment (juliendonck) [reply ui/Screen/DesignSystemGuide.svelte]: https://github.com/radicle-dev/radicle-upstream/pull/749/commits/f82d29b13a9575db8c724bf43b4772f4ed6f2a6d

File: public/typography.css [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+
+p,
+.typo-text {
+  font-family: var(--typeface-regular);
+  font-size: 16px;
+  line-height: 24px;
+}
+
+p.typo-bold,
+.typo-text-bold {
```
[the remark this answers is not in the document]
Comment (juliendonck) [reply public/typography.css:46]: Good point!

[the remark this answers is not in the document]
Comment (juliendonck) [reply public/typography.css]: https://github.com/radicle-dev/radicle-upstream/pull/749/commits/9ae15016c01093061bdf83260759347897817393

**Comment by juliendonck** [comment]
> As we push conncerns like layouting and colors into the individual componentsn do we plan to standardise on these in a similar refactoring in the future to avoid repetition and gurantee a consistent design language?

@xla yes, I think we can build on this with these modifiers like `typo-overflow-ellipsis`. @rudolfs and I are thinking about it. We need to see what makes sense here.

**Comment by xla** [comment]
> @xla yes, I think we can build on this with these modifiers like `typo-overflow-ellipsis`. @rudolfs and I are thinking about it. We need to see what makes sense here.

Great, Just wanted to make sure we see opportunity to take this further. This change itself is already a massive improvement.

**Comment by sarahscott** [comment]
<3 <3 <3

**Pull request merged**